File: services/user-service/src/main/java/com/visor/school/userservice/repository/PermissionRepository.java
package com.visor.school.userservice.repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.visor.school.userservice.model.Permission;

@Repository
public interface PermissionRepository extends JpaRepository<Permission, UUID> {
    Optional<Permission> findByPermissionKey(String permissionKey);
    boolean existsByPermissionKey(String permissionKey);
    List<Permission> findByCategory(String category);
}

File: services/user-service/src/main/kotlin/com/visor/school/userservice/repository/PermissionRepository.kt
package com.visor.school.userservice.repository

import com.visor.school.userservice.model.Permission
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional
import java.util.UUID

@Repository
interface PermissionRepository : JpaRepository<Permission, UUID> {
    fun findByPermissionKey(permissionKey: String): Optional<Permission>
    fun existsByPermissionKey(permissionKey: String): Boolean
    fun findByCategory(category: String): List<Permission>
}

